plugin.video.catchuptvandmore/resources/lib/channels/jp/tver.py

```diff
@@ -17,115 +17,104 @@
+        if program_title == program['content']['seriesTitle'] and program['content']['broadcasterName'] in BROADCASTER[item_id]:
+            item = Listitem()
+            item.label = '{} ({})'.format(program['content']['title'], program['content']['broadcastDateLabel'])
+            video_id = program['content']['id']
+            video_type = program['type']
+            item.art['thumb'] = item.art['landscape'] = URL_VIDEO_PICTURE.format(video_type, video_id)
+            resp = urlquick.get(URL_VIDEOS_REPLAY.format(video_type, video_id),
+                                headers={'User-Agent': web_utils.get_random_ua()},
+                                max_age=-1)
+            data = resp.json()
+            item.info['plot'] = data['description']
+            item.set_callback(get_video_url, data)
+            item_post_treatment(item)
+            yield item
 
 
 @Resolver.register
-def get_video_url(plugin,
-                  item_id,
-                  video_url,
-                  download_mode=False,
-                  **kwargs):
+def get_video_url(plugin, data, **kwargs):
 
     is_helper = inputstreamhelper.Helper('mpd', drm='widevine')
     if not is_helper.check_inputstream():
         return False
 
-    resp = urlquick.get(video_url,
-                        headers={'User-Agent': web_utils.get_random_ua()},
-                        max_age=-1)
-    stream_datas = resp.text.split('addPlayer(')[1].split(');')[0].replace(
-        "\n", "").replace("\r", "").split(',')
-    data_account = stream_datas[0].strip().replace("'", "")
-    data_player = stream_datas[1].strip().replace("'", "")
-    if item_id == 'tx':
-        data_video_id = stream_datas[4].strip().replace("'", "")
+    if 'videoID' in data['video']:
+        data_video = data['video']['videoID']
+        ref = ''
     else:
-        data_video_id = 'ref:' + stream_datas[4].strip().replace("'", "")
+        data_video = data['video']['videoRefID']
+        ref = 'ref:'
+    data_account = data['video']['accountID']
+    data_player = data['video']['playerID']
 
     # Method to get JSON from 'edge.api.brightcove.com'
-    resp = urlquick.get(
-        URL_BRIGHTCOVE_VIDEO_JSON % (data_account, data_video_id),
-        headers={
-            'User-Agent':
-            web_utils.get_random_ua(),
-            'Accept':
-            'application/json;pk=%s' %
-            (get_brightcove_policy_key(data_account, data_player)),
-            'X-Forwarded-For':
-            plugin.setting.get_string('header_x-forwarded-for')
-        })
-    json_parser = json.loads(resp.text)
+    file_js = urlquick.get(URL_BRIGHTCOVE_POLICY_KEY % (data_account, data_player))
+    policy = re.compile('policyKey:"(.+?)"').findall(file_js.text)[0]
+    headers = {
+        'User-Agent': web_utils.get_random_ua(),
+        'Accept': 'application/json;pk=%s' % policy,
+        'X-Forwarded-For': plugin.setting.get_string('header_x-forwarded-for')
```

Comment:
We don't want add-ons to circumvent GEO blocks, both in streaming and downloading. That means that using `x-forwarded-for` is not allowed.